### PR TITLE
feat(web): digest workflow tour - hints ui

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

Digest Onboarding Workflow - the tour hints UI. **PART 2.**

### Why was this change needed?

This PR is a part of improvements over the onboarding process.
### Other information (Screenshots)

Screenshot 2023-03-11 at 14 05 41

Screenshot 2023-03-11 at 14 06 31

https://user-images.githubusercontent.com/2607232/224492824-345c1ccc-81a2-4d20-906d-691b72093460.mov

https://user-images.githubusercontent.com/2607232/224492833-19f41f18-d1b7-40eb-9bc8-84a70a2a08d1.mov

### CODE REVIEW

File: apps/web/cypress/tests/digest-playground.spec.ts [hunk 1]
```diff
@@ -32,4 +32,103 @@ describe('Digest Playground Workflow Page', function () {
 
     cy.url().should('include', '/templates/edit');
   });
+
+  it('should show the digest workflow hints', () => {
```
Comment (LetItRock) [inline apps/web/cypress/tests/digest-playground.spec.ts:36]: the e2e test that verifies the whole flow:
--> from the templates page --> digest playground --> workflow editor --> all digest workflow hints

File: apps/web/cypress/tests/digest-playground.spec.ts [hunk 2]
```diff
@@ -32,4 +32,103 @@ describe('Digest Playground Workflow Page', function () {
...
+    // the step settings should be hidden
+    cy.getByTestId('drag-side-menu').contains('Steps to add');
+
+    // click got it should hide the hint
+    cy.getByTestId('digest-workflow-tooltip-primary-button').contains('Got it').click();
+
+    cy.getByTestId('digest-workflow-tooltip').should('not.exist');
+  });
+
+  it('should hide the digest workflow hints when clicking on skip tour button', () => {
```
Comment (LetItRock) [inline apps/web/cypress/tests/digest-playground.spec.ts:109]: this one verifies the skip tour button

File: apps/web/src/design-system/button/Button.styles.ts [hunk 3]
```diff
@@ -45,6 +45,10 @@ export default createStyles(
         '&&:hover': {
           backgroundSize: '100%',
         },
+        '&:focus': {
+          outlineOffset: 0,
+          outline: '2px solid rgb(153,200,255)',
+        },
```
Comment (LetItRock) [inline apps/web/src/design-system/button/Button.styles.ts:51]: By default the tour library that we do use supports the keyboard navigation, this line is a style for the focused button. I've made it look the same styles as the mantine has for the unstyled button (skip tour button). I couldn't find in the design system how it supposed to look, but I know that we will change it in the near future as Nikolay wanted to improve the components.

File: apps/web/src/design-system/dots-navigation/DotsNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    transition: transform 0.3s ease;
+  }
+
+  &:hover,
+  &:focus {
+    opacity: 0.7;
+  }
+`;
+
+export const DotsNavigation = ({
```
Comment (LetItRock) [inline apps/web/src/design-system/dots-navigation/DotsNavigation.tsx:59]: the dots navigation component used in the hints... I've added it to our design system
Screenshot 2023-03-11 at 16 30 57

File: apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  gap: 28px;
+  margin-top: 24px;
+`;
+
+const DotsNavigationStyled = styled(DotsNavigation)`
+  margin-top: 24px;
+  align-self: center;
+`;
+
+export const DigestWorkflowTourTooltip = ({
```
Comment (LetItRock) [inline apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx:55]: The digest workflow tour hint component contains the logic for three steps. 
The texts and icons are switched depending on the tour step index.

File: apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    if (tourStepIndex === 1) {
+      const emailStep = steps.find((el) => el.template?.type === StepTypeEnum.EMAIL);
+      setStep(tourStepIndex);
+      setSelectedNodeId(emailStep?.id || '');
+    }
+    if (tourStepIndex === 2) {
+      setStep(tourStepIndex);
+      setSelectedNodeId('');
+    }
+  };
```
Comment (LetItRock) [inline apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx:85]: The callback is triggered when clicking on the dots navigation item or the next button. 
The logic to select the next tour step and highlight the workflow editor node.

File: apps/web/src/design-system/dots-navigation/DotsNavigation.tsx (hunk 4 above)
Comment (djabarovgeorge) [reply apps/web/src/design-system/dots-navigation/DotsNavigation.tsx:59]: Amazing I will add it to preview page as well 😁

File: apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+import { useTour } from './TourProvider';
+import { Button, colors, DotsNavigation } from '../../../design-system';
+import { Clock, LetterOpened, BellWithNotification } from '../../../design-system/icons';
+import { useTemplateEditorContext } from './TemplateEditorProvider';
+import { IForm } from '../components/formTypes';
+
+const ICONS = [Clock, LetterOpened, BellWithNotification];
+const TITLE = ['Set-up time interval', 'Set-up email content', 'Test your workflow'];
+const DESCRIPTION = [
```
Comment (djabarovgeorge) [inline apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx:15]: Just a thought not a must at all, what do you think about extracting those to a separate file, we could add other tour data there as well, in addition, what do you think about merging them to an object list with properties of title and description?

File: apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+  const steps = watch('steps');
+  const { stopTour, setStep } = useTour();
+
+  const Icon = ICONS[index];
+
+  const handleOnClick = (tourStepIndex: number) => {
+    if (tourStepIndex === 0) {
+      const digestStep = steps.find((el) => el.template?.type === StepTypeEnum.DIGEST);
+      setStep(tourStepIndex);
+      setSelectedNodeId(digestStep?.id || '');
```
Comment (djabarovgeorge) [inline apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx:74]: What do you think about making a 'return' call at the end of the if statements?

**Review by djabarovgeorge — APPROVED** [review]
Looks good to me :)

File: apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx (hunk 7 above)
Comment (LetItRock) [reply apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx:15]: this component has a specific digest tour logic, the only thing we can do is move the body to the design system... also the tour library requires you to pass the component like this, so you can't pass the props this way... 
```
const digestTourSteps: Step[] = [
  {...
    tooltipComponent: DigestWorkflowTourTooltip,
}...
```
so if it is specific to the digest tour then I feel it's fine to leave it like it is... also merging to the object won't have here any benefits...

File: apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx (hunk 8 above)
Comment (LetItRock) [reply apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx:74]: I'll change it to `else if`

**Pull request merged**